code/datums/actions/actions.dm

```diff
@@ -299,6 +302,9 @@
 			return
 		owner.visible_message(SPAN_NOTICE("[owner] assembles \the [obj_name]!"))
 		var/obj/item/R = new obj_type(obj_turf)
+		if (!R.density && !src.constructible_check(R) && ishuman(owner))
```

Comment:
saner to just do an `isitem()` check, no?

Reply:
Very true.